lib/services/search.js

```diff
@@ -0,0 +1,395 @@
+'use strict';
+
+const es = require('elasticsearch'),
+  bluebird = require('bluebird'),
+  _ = require('lodash'),
+  util = require('util'),
+  log = require('./log').withStandardPrefix(__filename),
+  endpoint = process.env.ELASTIC_HOST || '',
+  serverConfig = {
+    host: endpoint,
+    maxSockets: 500,
+    apiVersion: '1.7',
+    // log: ['error', 'trace'],
+    defer: require('../utils/defer')
+  };
+
+/**
+ * Test whether or not the client is connected
+ * to Elastic
+ *
+ * @param  {object} currentClient
+ * @returns {Promise}
+ */
+function healthCheck(currentClient) {
+  return currentClient.ping({
+    requestTimeout: 1000
+  }).then(function () {
+    log('info', 'Elasticsearch cluster is up!');
+    return bluebird.resolve();
+  }).catch(function (error) {
+    log('info', 'Elasticsearch cluster is down!');
+    return bluebird.reject(error);
+  });
+}
+
+/**
+ * Create an Elasticsearch index
+ *
+ * @param {string} index
+ * @param {object} mappings
+ * @returns {Promise}
+ */
+function initIndex(index, mappings) {
+  return module.exports.client.indices.create({
+    index: index,
+    body: mappings
+  }).then(function () {
+    log('info', 'Successfully created ' + index);
+  })
+    .catch(function (error) {
+      log('error', error);
+    });
+}
+
+/**
+ * Check if an Elasticsearch index exists
+ *
+ * @param {string} index
+ * @returns {Promise}
+ */
+function existsIndex(index) {
+  return module.exports.client.indices.exists({
+    index: index
+  });
+}
+
+/**
+ * Indices should have a different name than the alias
+ * Append '_v1' to newly created indices
+ * @param {string} alias
+ * @returns {string}
+ */
+function createIndexName(alias) {
+  return alias + '_v1';
```

Comment:
yes we still need aliases. aliases are useful when reindexing and you don't want to lose any data because you can point an index (published-articles_v1) to an alias (published-articles). see here for more info https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-aliases.html